examples/cpp/Breakpoint.cc

```diff
@@ -0,0 +1,156 @@
+/*
+ * This is an example of a hardware breakpoint on a kernel address.
+ * run in project examples directory with:
+ * sudo ./breakpoint.py <0xaddress> <pid> <breakpoint_type>
+ * HW_BREAKPOINT_W = 2
+ * HW_BREAKPOINT_RW = 3
+ *
+ * You may need to clear the old tracepipe inputs before running the script : 
+ * echo > /sys/kernel/debug/tracing/trace 
+ *
+ *  10-Jul-2019   Aanandita Dhawan   Created this.
+ */
+
+
+#include <unistd.h>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <stdlib.h>
+#include <stddef.h>
+#include <algorithm>
+#include "BPF.h"
+#include <inttypes.h>
+#include <stdio.h>
+#include <iostream>
+#include <string>
+
+const std::string BPF_PROGRAM = R"(
+
+#include <linux/sched.h>
+#include <uapi/linux/ptrace.h>
+
+struct stack_key_t {
+  int pid;
+  char name[16];
+  int user_stack;
+  int kernel_stack;
+};
+
+BPF_STACK_TRACE(stack_traces, 16384);
+BPF_HASH(counts, struct stack_key_t, uint64_t);
+
+int func(struct pt_regs *ctx) {
+  struct stack_key_t key = {};
+  key.pid = bpf_get_current_pid_tgid() >> 32;
+  bpf_get_current_comm(&key.name, sizeof(key.name));
+  key.kernel_stack = stack_traces.get_stackid(ctx, 0);
+  key.user_stack = stack_traces.get_stackid(ctx, BPF_F_USER_STACK);
+
+  u64 zero = 0, *val;
+  val = counts.lookup_or_init(&key, &zero);
+  (*val)++;
+
+  bpf_trace_printk("Hello World, Here I accessed the address!\n");
+
+  return 0;
+
+}
+
+)";
+
+struct stack_key_t {
+  int pid;
+  char name[16];
+  int user_stack;
+  int kernel_stack;
+};
+
+int main(int argc, char** argv) {
+  
+  ebpf::BPF bpf;
+  auto init_res = bpf.init(BPF_PROGRAM);
+  if (init_res.code() != 0) {
+    std::cerr << init_res.msg() << std::endl;
+    return 1;
+  }
+
+  std::ifstream pipe("/sys/kernel/debug/tracing/trace_pipe");
+  std::string line;
+  
+  uint64_t symbol_addr; 
+  char *end; 
+  long int num;
```

Comment:
symbol_addr is not used and num is a bad var name for addr. Maybe just use symbol_addr.